assignment7.txt

```diff
@@ -0,0 +1,16 @@
+1. For an array {1, 2, 3, 4}
+	a. ++*p will get the variable located at *p then increment it. if *p is at index 0, new array would be {2, 2, 3, 4} with *p at index 0
```

Comment:
The expression ++*p has two operators of same precedence, so compiler looks for associativity. Associativity of operators is right to left. Therefore the expression is treated as ++(*p). The expression *p++ is treated as *(p++) as the precedence of postfix ++ is higher than *. The expression *++p has two operators of same precedence, so compiler looks for associativity. Associativity of operators is right to left. Therefore the expression is treated as *(++p).